src/wovnio/wovnphp/Headers.php

```diff
@@ -63,15 +63,20 @@ public function __construct(&$env, &$store)
         $this->unmaskedUrl = $this->protocol . '://' . $this->unmaskedHost . $this->unmaskedPathname;
         $this->host = $this->unmaskedHost;
         if ($store->settings['url_pattern_name'] === 'subdomain') {
-            $this->host = $this->removeLang($this->host, $this->lang());
+            $intermediateHost = explode('//', $this->removeLang($this->protocol . '://' . $this->host, $this->lang()));
```

Comment:
What was the motivation behind adding protocol to it? So `removeLang` can always assume the parameter is a full URI? I notice below this we call removeLang with just the query, though.

Reply:
Yep, this is to fix a bug I discovered related to super **malformed href links when using subdomain pattern**. The remove_lang logic can only correctly handle subdomain patterns when the input is in the `http://english.example.com` form. I tried to find a root cause for this behavior but couldn't. Using this as a temporary fix I guess.